packages/client-config/API.md

```diff
@@ -208,7 +208,7 @@ export const DEFAULT_CLIENT_CONFIG_VERSION: ClientConfigVersion;
 export const generateClientConfig: <T extends "1" | "0">(credentialProvider: AwsCredentialIdentityProvider, backendIdentifier: DeployedBackendIdentifier, version: T) => Promise<ClientConfigVersionTemplateType<T>>;
 
 // @public
-export const generateClientConfigToFile: (credentialProvider: AwsCredentialIdentityProvider, backendIdentifier: DeployedBackendIdentifier, version: ClientConfigVersion, outDir?: string, format?: ClientConfigFormat, log?: ((message: string) => void) | undefined) => Promise<void>;
+export const generateClientConfigToFile: (credentialProvider: AwsCredentialIdentityProvider, backendIdentifier: DeployedBackendIdentifier, version: ClientConfigVersion, outDir?: string, format?: ClientConfigFormat, log?: ((message: string) => void) | undefined, debug?: ((message: string) => void) | undefined) => Promise<void>;

```

Comment:
This is not great.

Alternatives:
1. Un-static this, get `type` and implementing `class` +/- factory. And make `debug` optional ctor/factory parameter (in case we want to somewhat preserve this design).
2. Invest in `Logger` abstraction and have `Printer` become implementation of "logging strategy". I.e. `Printer` should never be promoted above `cli-core` (i.e. cli-like package) - this is because it's no longer "printing" but "logging". You can check how popular logging libraries achieve this (like log4j in java or winston in JS). See also this attempt https://github.com/aws-amplify/amplify-backend/pull/1115#discussion_r1520463569 .
3. Instead of injecting logger make return type of these functions complex type, so that additional informaton is passed to "cli-like" packages so they can print relevant info. (see `fsPromises.mkdir` for example - it's return indicates what happened).

Reply:
My opinion is that option 3 is probably the min-cost solution for now. (And may even be the best long term solution if this package is intended to be a standalone library)

Reply:
I'm wondering what the distinction is between logging and using `printer.log` with `LogLevel.DEBUG`?

I'll go ahead and work on option 3.

Reply:
thanks.
I'd say that `log?:` parameter is also unfortunate and should be refactored. But this wasn't added in this PR, so might be out of scope (unless you want to address it).